openssl/apibridge_1_1.c

```diff
@@ -109,58 +72,14 @@ local_HMAC_CTX_reset(HMAC_CTX* ctx) {
     go_openssl_HMAC_CTX_init(ctx);
 }
 
-struct md5_sha1_ctx {
-  MD5_CTX md5;
-  SHA_CTX sha1;
-};
-
-static int
-md5_sha1_init(EVP_MD_CTX *ctx)
-{
-  struct md5_sha1_ctx *mctx = go_openssl_EVP_MD_CTX_md_data(ctx);
-  if (!go_openssl_MD5_Init(&mctx->md5))
-    return 0;
-  return go_openssl_SHA1_Init(&mctx->sha1);
-}
-
-static int md5_sha1_update(EVP_MD_CTX *ctx, const void *data, size_t count)
-{
-  struct md5_sha1_ctx *mctx = go_openssl_EVP_MD_CTX_md_data(ctx);
-  if (!go_openssl_MD5_Update(&mctx->md5, data, count))
-    return 0;
-  return go_openssl_SHA1_Update(&mctx->sha1, data, count);
-}
-
-static int md5_sha1_final(EVP_MD_CTX *ctx, unsigned char *md)
-{
-  struct md5_sha1_ctx *mctx = go_openssl_EVP_MD_CTX_md_data(ctx);
-  if (!go_openssl_MD5_Final(md, &mctx->md5))
-    return 0;
-  return go_openssl_SHA1_Final(md + MD5_DIGEST_LENGTH, &mctx->sha1);
-}
-
-// Change: Removed:
-// static int ctrl(EVP_MD_CTX *ctx, int cmd, int mslen, void *ms)
-
-static const EVP_MD md5_sha1_md = {
-    NID_md5_sha1,
-    NID_md5_sha1,
-    MD5_DIGEST_LENGTH + SHA_DIGEST_LENGTH,
-    0,
-    md5_sha1_init,
-    md5_sha1_update,
-    md5_sha1_final,
-    NULL,
-    NULL,
-    EVP_PKEY_NULL_method, // Change: inserted
-    MD5_CBLOCK,
-    sizeof(EVP_MD *) + sizeof(struct md5_sha1_ctx),
-    NULL, // Change: was ctrl
-};
-
 const EVP_MD* local_EVP_md5_sha1(void)
 {
-  return &md5_sha1_md;
+    // MD5SHA1 is not implemented in OpenSSL 1.0.2.
+    // It is implemented in higher versions but without FIPS support.
+    // It is considered a deprecated digest, not approved by FIPS 140-2
+    // and only used in pre-TLS 1.2, so we would rather not support it
+    // if using 1.0.2 than than implement something that is not properly validated.
+  return NULL;
```

Comment:
Where do we document that SHA1 isn't supported in OpenSSL 1.0.2? Feel like there should be a doc change accompanying this.

Reply:
This module does not have much documentation at the moment. I will submit a PR with some minimal documentation once the OpenSSL 3 are merged, which would include this fact.

Once our Go toolchain upgrades this module I will also update the detailed documentation in there.